packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -1,80 +1,235 @@
+    const modalHeaderId = 'modal-header-verify-email';
+    const modalDescId = 'modal-desc-verify-email';
 
-  return (
+    const resendPrimaryEmailCodeFromModal = useCallback(() => {
+      console.log('resend primary email verification code - part of FXA-1613');
+      hideModal();
+      revealAlertBar();
+    }, [hideModal]);
+
+    const modalTriggerElement = useRef<HTMLButtonElement>(null);
+    // If the UnitRow children contains an AlertBar that is revealed,
+    // don't redirect focus back to the element that opened the modal
+    // because focus will be set in the AlertBar.
+    useFocusOnTriggeringElementOnClose(
+      modalRevealed,
+      modalTriggerElement,
+      alertBarRevealed
+    );
+    const route = primaryEmailIsVerified
+      ? '/beta/settings/secondary_email'
+      : undefined;
+    const revealModalIfPrimaryEmailIsNotVerified = primaryEmailIsVerified
+      ? undefined
+      : revealModal;
+
+    // user doesn't have a secondary email (verified or unverified) set
+    return (
       <UnitRow
         header="Secondary email"
         headerValue={null}
+        revealModal={revealModalIfPrimaryEmailIsNotVerified}
         {...{
-          revealModal,
+          route,
           modalRevealed,
-          alertBarRevealed
+          alertBarRevealed,
         }}
       >
-        <p className="text-sm mt-3">
-          Access your account if you can't log in to your primary email.
-        </p>
-        <p className="text-grey-400 text-xs mt-2">
-          Note: a secondary email won't restore your information—you'll need
-          a{' '}
-          <a className="link-blue" href="#recovery-key">
-            recovery key
-          </a>{' '}
-          for that.
-        </p>
+        <SecondaryEmailDefaultContent />
 
         {modalRevealed && (
           <Modal
             onDismiss={hideModal}
-            onConfirm={onSecondaryEmailConfirm}
+            onConfirm={resendPrimaryEmailCodeFromModal}
             headerId={modalHeaderId}
             descId={modalDescId}
           >
             <h2
               id={modalHeaderId}
               className="font-bold text-xl text-center mb-2"
+              data-testid={modalHeaderId}
             >
               Verify primary email first
             </h2>
-            <p className="text-center" id={modalDescId}>
-              Before you can add a secondary email, you must verify your
-              primary email. To do this, you'll need access to{' '}
-              {primaryEmail}
+            <p
+              className="text-center"
+              id={modalDescId}
+              data-testid={modalDescId}
+            >
+              Before you can add a secondary email, you must verify your primary
+              email. To do this, you'll need access to {primaryEmail}
             </p>
           </Modal>
         )}
+      </UnitRow>
+    );
+  };
+
+  const UnitRowSecondaryEmailContentAndActions = ({
```

Comment:
Thought: since we are already in the "secondary email" context because of the component name, prefixing everything with "secondary" could be redundant. You could call the main record object `secondaryEmail` and destructure the two variables below down to `const { email, verified } = secondaryEmail;` without fear of namespace conflict. It also allows the variable names to be much shorter, less code overall, and you could use that sweet, sweet shorthand property syntax in the query on L121.

This idea could actually be implemented in all the areas that prefix the word secondary onto variables that exist inside an already-secondary context.

Reply:
Ugh yeah, my problem with that is that this component also deals with the primary email in some instances, like checking `primaryEmailIsVerified`, showing the primary email in the alert bar, and resending the primary email verification code (done in a later task). I honestly feel like being explicit here is going to be much nicer when in a couple of years someone who's never touched this before can look at the file and tell what's going on without ambiguity. I'm a fan of shorter names, I just have this gut feeling that it's better to be overly specific here to help with any potential confusion later on if something is referring to primary or secondary. Whatchu think?